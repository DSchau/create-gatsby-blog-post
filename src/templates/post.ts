const frontmatter = ({ date, folder, capitalized, tags }) => {
  return [
    '___',
    `path: "/${folder}.html"`,
    `date: "${date.toJSON()}"`,
    `title: "${capitalized}"`,
    tags && 'tags: []',
    '___'
  ]
    .filter(val => val && val.length > 0)
    .join('\n');
};

export default function post(options) {
  return [
    frontmatter(options),
    null,
    'Your next, great blog post goes here',
    null
  ].join('\n');
}
