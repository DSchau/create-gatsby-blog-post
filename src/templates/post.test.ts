import post from './post';

describe('post template', () => {
  const date = new Date('10/08/1990');
  const postObj = {
    capitalized: 'Hello World',
    date,
    folder: 'hello-world',
    tags: true
  };

  it('writes expected template', () => {
    expect(post(postObj)).toMatchSnapshot();
  });
});
