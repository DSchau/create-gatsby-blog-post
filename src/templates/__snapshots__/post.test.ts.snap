// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post template writes expected template 1`] = `
"---
path: \\"/hello-world\\"
date: \\"1990-10-08T05:00:00.000Z\\"
title: \\"Hello World\\"
tags: []
---

Your next, great blog post goes here
"
`;
